src/test/java/com/faforever/api/clan/ClanControllerIntegrationTest.java

```diff
@@ -81,205 +56,159 @@ public void setUp() {
-                .param("username", login)
-                .param("password", password)
-                .param("grant_type", "password"));
-    auth.andExpect(status().isOk());
-    JsonNode node = objectMapper.readTree(auth.andReturn().getResponse().getContentAsString());
-    return "Bearer " + node.get("access_token").asText();
-  }
-
-  private Player createPlayer(String login) throws Exception {
-    User user = (User) new User()
-        .setPassword("foo")
-        .setLogin(login)
-        .setEmail(login + "@faforever.com");
-    userRepository.save(user);
-    return playerRepository.findOne(user.getId());
-  }
-
-
   @Test
   public void meDataWithoutClan() throws Exception {
-    String accessToken = createUserAndGetAccessToken("Dragonfire", "foo");
+    Session session = SessionFactory.createUserAndGetAccessToken(
+        database, mvc);
+
     String expected = String.format("{\"player\":{\"id\":%s,\"login\":\"%s\"},\"clan\":null}",
-        me.getId(),
-        me.getLogin());
+        session.getPlayer().getId(),
+        session.getPlayer().getLogin());
 
-    assertEquals(1, playerRepository.count());
+    assertEquals(1, database.getPlayerRepository().count());
     this.mvc.perform(get("/clans/me/")
-        .header("Authorization", accessToken))
+        .header("Authorization", session.getToken()))
         .andExpect(content().string(expected))
         .andExpect(status().isOk());
   }
 
   @Test
   public void meDataWithClan() throws Exception {
-    String accessToken = createUserAndGetAccessToken("Dragonfire", "foo");
+    Session session = SessionFactory.createUserAndGetAccessToken(
+        database, mvc);
 
-    Clan clan = new Clan().setLeader(me).setTag("123").setName("abcClanName");
-    ClanMembership myMembership = new ClanMembership().setPlayer(me).setClan(clan);
+    Clan clan = new Clan().setLeader(session.getPlayer()).setTag("123").setName("abcClanName");
+    ClanMembership myMembership = new ClanMembership().setPlayer(session.getPlayer()).setClan(clan);
     clan.setMemberships(Collections.singletonList(myMembership));
-    clanRepository.save(clan);
+    database.getClanRepository().save(clan);
 
     String expected = String.format("{\"player\":{\"id\":%s,\"login\":\"%s\"},\"clan\":{\"id\":%s,\"tag\":\"%s\",\"name\":\"%s\"}}",
-        me.getId(),
-        me.getLogin(),
+        session.getPlayer().getId(),
+        session.getPlayer().getLogin(),
         clan.getId(),
         clan.getTag(),
         clan.getName());
 
-    assertEquals(1, playerRepository.count());
-    assertEquals(1, clanRepository.count());
-    assertEquals(1, clanMembershipRepository.count());
+    assertEquals(1, database.getPlayerRepository().count());
+    assertEquals(1, database.getClanRepository().count());
+    assertEquals(1, database.getClanMembershipRepository().count());
     this.mvc.perform(get("/clans/me/")
-        .header("Authorization", accessToken))
+        .header("Authorization", session.getToken()))
```

Comment:
I though about this for while ....
It's hard because you have so many `versions` of super methods ...
In c# you can simply overload an existing method of an class ...